### ISSUE TITLE
Access one container from another one?

### ISSUE THREAD

**Issue by goloroden** [docker/for-aws#12]
Supposed I run three containers on Docker for AWS.

Two if them contain a web server that expose their content on ports 3000 and 4000.
Now I want both of them to be reachable from the outside on port 80.

Obviously, I can not expose port 80 on *both* of them, so I need some kind of reverse proxy (such as Nginx). This is my third container.

Now, within Nginx I need to configure an upstream server. How do I know the IP address of the other two containers? Respectively, is there some possibility to access them by host name (which would be better, IMHO)?

In other words: When running on Docker for AWS, can one container find another one by its hostname?

My first approach was simply to run an interactive sample container using bash, and trying to `ping` one of the other ons by their hostname, but without luck. Is this possible, and if so, how?

**Comment by kencochrane** [comment]
@goloroden Here is how I would do it. 

First off I would [create a overlay network](https://docs.docker.com/engine/swarm/networking/) for the services to talk over.

```
docker network create --driver overlay web-network
```

Now I will create 3 services, one for the ones you described

```
docker service create --name web3000 -p 3000 --network web-network myname/web3000

docker service create --name web4000 -p 4000 --network web-network myname/web4000

docker service create --name webnginx -p 80:80 --network web-network myname/nginx
```

Now all 3 services will be installed on the swarm, and they will talk across the `web-network` overlay network. If you want to connect to the `web3000` and `web4000` services from webnginx you just need to use their name. `web3000` and `web4000` and swarm will automatically connect you to the service. If you have more than one container running, it will also [load balance](https://docs.docker.com/engine/swarm/key-concepts/#/load-balancing) the requests between the containers for you, automatically.

Then all you need to do is call the ELB address that comes with docker for AWS, and it should connect you to the webnginx service from the outside network.

**Comment by goloroden** [comment]
Great, thanks for the info, this helps 👍
